### PR TITLE
Enable GC logs for builds as CircleCI artifacts

### PR DESCRIPTION
**Pull request by schlosna**
Adds GC logs to CircleCI artifacts, e.g. `build-2017-01-04_14-51-31-pid11879.gc.log`

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/1398)

### CODE REVIEW

**Comment by schlosna** [comment]
Example output: https://5623-36960637-gh.circle-artifacts.com/0/tmp/circle-artifacts.ty67MHX/heapdumps/build-2017-01-04_15-02-23-pid24367.gc.log

File: scripts/circle-ci/run-circle-tests.sh [hunk 1]
```diff
@@ -39,16 +39,25 @@ if ./scripts/circle-ci/check-only-docs-changes.sh; then
     exit 0
 fi
 
+JAVA_GC_LOGGING_OPTIONS="${JAVA_GC_LOGGING_OPTIONS} -XX:+HeapDumpOnOutOfMemoryError"
+JAVA_GC_LOGGING_OPTIONS="${JAVA_GC_LOGGING_OPTIONS} -verbose:gc"
+JAVA_GC_LOGGING_OPTIONS="${JAVA_GC_LOGGING_OPTIONS} -XX:+PrintGC"
+JAVA_GC_LOGGING_OPTIONS="${JAVA_GC_LOGGING_OPTIONS} -XX:+PrintGCDateStamps"
+JAVA_GC_LOGGING_OPTIONS="${JAVA_GC_LOGGING_OPTIONS} -XX:+PrintGCDetails"
+JAVA_GC_LOGGING_OPTIONS="${JAVA_GC_LOGGING_OPTIONS} -XX:-TraceClassUnloading"
+JAVA_GC_LOGGING_OPTIONS="${JAVA_GC_LOGGING_OPTIONS} -Xloggc:build-%t-%p.gc.log"
```
Comment (jeremyk-91) [inline scripts/circle-ci/run-circle-tests.sh]: nit: since this should be run in bash I think `+=` would be more readable here.

Comment (schlosna) [inline scripts/circle-ci/run-circle-tests.sh]: There is only [bourne shell](https://en.wikipedia.org/wiki/Bourne_shell)

GC logging overhead is negligible and we run it in production every where that matters, so I think our builds can handle it.

Comment (jeremyk-91) [inline scripts/circle-ci/run-circle-tests.sh]: Okay. The script does begin with `#!/usr/bin/env bash` though, and it's used later on with the `BASE_GRADLE_ARGS` so I'm pretty confident it's fine to use `+=`, unless I'm missing something?

I'm happy with the change in any case.

Comment (schlosna) [inline scripts/circle-ci/run-circle-tests.sh]: I generally try to avoid bashisms when possible

**Pull request merged**